### ISSUE TITLE
rpath is not set for all binaries

### ISSUE THREAD

**Issue by BMBurstein** [stefanberger/swtpm#799]
when using libtpms that is not in a standard path, and passing it to autogen via `LIBTPMS_LIBS`, not all the resulting binaries work. `swtpm` and `swtpm_cuse` work, while all the others don't. They fail to start with an error about not finding `libtpms.so.0`. Upon investigation it seems that the rpath is only set for those 2 binaries, and not for the rest or for `libswtpm_libtpms.so`.

I don't know much about automake/libtool, but it seems that in the final Makefile in src/swtpm there is this rule:

```
libswtpm_libtpms.la: $(libswtpm_libtpms_la_OBJECTS) $(libswtpm_libtpms_la_DEPENDENCIES) $(EXTRA_libswtpm_libtpms_la_DEPENDENCIES) 
	$(AM_V_CCLD)$(libswtpm_libtpms_la_LINK) -rpath $(privlibdir) $(libswtpm_libtpms_la_OBJECTS) $(libswtpm_libtpms_la_LIBADD) $(LIBS)
```

I don't know how it gets there, but I assume it is responsible for the rpath being correct here.

**Comment by stefanberger** [comment]
> when using libtpms that is not in a standard path

What is non-standard path for a library? I would expect that `ldconfig -v` has to show it. Does it show it?

**Comment by BMBurstein** [comment]
Sorry, I tried testing it again and realized I had misunderstood my problem. I am not sure it is even a problem.
I am passing autogen.sh a prefix as well. The prefix is the same as the custom path for libtpms,so, which is why I got confused. It appears that none of the binaries remember where the libtpms.so is. `swtpm` and `swtpm_cuse` do, however, add the prefix (actually, the libdir) to the rpath, probably because they use `libswtpm_libtpms.so`, which by chance in my case allowed finding the libtpms.so
